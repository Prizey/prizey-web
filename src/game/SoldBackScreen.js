import React from 'react'
import { Redirect } from '@reach/router'
import { Button, Typography } from '@material-ui/core'

import UserBalance from 'design/UserBalance'
import Layout from 'design/Layout/Layout'
import ProfileLink from 'design/ProfileLink/ProfileLink'

const SoldBackComponent = ({ navigate, currentUser, location }) => (
  <Layout
    location={location}
    currentUser={currentUser}
    leftIcon={<UserBalance />}
    rightIcon={<ProfileLink />}
  >
    <Typography align="center" variant="h5">
      Congrats! You sold the product for 3 diamonds.
    </Typography>
    <br />
    <Button
      variant="contained"
      color="secondary"
      fullWidth
      aria-label="Go to home"
      onClick={() => {
        navigate('/')
      }}
    >
      Go to home
    </Button>

    <Button
      variant="contained"
      color="primary"
      fullWidth
      aria-label="Play Again"
      onClick={() => {
        navigate('/game')
      }}
    >
      Play Again
    </Button>
  </Layout>
)

export default ({ currentUser, ...props }) =>
  currentUser ? (
    <SoldBackComponent {...props} />
  ) : (
    <Redirect to="/sign-in?next=/sold-back" noThrow />
  )
