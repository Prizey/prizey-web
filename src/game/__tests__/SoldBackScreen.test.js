import React from 'react'
import renderer from 'react-test-renderer'
import SoldBackScreen from '../SoldBackScreen'

jest.mock('design/UserBalance', () => () => null)

it('renders correctly', () => {
  const tree = renderer.create(<SoldBackScreen currentUser />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('redirects correctly', () => {
  const tree = renderer.create(<SoldBackScreen />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('navigate to home when click on Go to home', () => {
  const params = {
    currentUser: true,
    navigate: jest.fn(),
  }
  const tree = renderer.create(<SoldBackScreen {...params} />).root

  tree.findByProps({ 'aria-label': 'Go to home' }).props.onClick()
  expect(params.navigate).toHaveBeenCalledWith('/')
})

it('navigate to game when click on Play Again', () => {
  const params = {
    currentUser: true,
    navigate: jest.fn(),
  }
  const tree = renderer.create(<SoldBackScreen {...params} />).root

  tree.findByProps({ 'aria-label': 'Play Again' }).props.onClick()
  expect(params.navigate).toHaveBeenCalledWith('/game')
})
