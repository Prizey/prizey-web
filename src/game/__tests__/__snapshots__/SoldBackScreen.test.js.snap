// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redirects correctly 1`] = `null`;

exports[`renders correctly 1`] = `
<div
  className="Component-root-8"
>
  <div
    className="Component-header-6"
  />
  <div
    className="Component-container-3"
  >
    <img
      alt="Prizey"
      className="Component-logoImage-7"
      src="/logo.png"
    />
    <div
      className="Component-caption-1"
    />
    <div
      className="Component-children-2"
    >
      <h5
        className="MuiTypography-root-9 MuiTypography-h5-25 MuiTypography-alignCenter-32"
      >
        Congrats! You sold the product for 3 diamonds.
      </h5>
      <br />
      <button
        aria-label="Go to home"
        className="MuiButtonBase-root-71 MuiButton-root-45 MuiButton-contained-56 MuiButton-containedSecondary-58 MuiButton-raised-59 MuiButton-raisedSecondary-61 MuiButton-fullWidth-70"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-46"
        >
          Go to home
        </span>
        <span
          className="MuiTouchRipple-root-74"
        />
      </button>
      <button
        aria-label="Play Again"
        className="MuiButtonBase-root-71 MuiButton-root-45 MuiButton-contained-56 MuiButton-containedPrimary-57 MuiButton-raised-59 MuiButton-raisedPrimary-60 MuiButton-fullWidth-70"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-46"
        >
          Play Again
        </span>
        <span
          className="MuiTouchRipple-root-74"
        />
      </button>
    </div>
    <div
      className="Component-footer-4"
    >
      <a
        className="Component-footerLink-5"
        href="https://termly.io/en/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <span
          className="MuiTypography-root-9 MuiTypography-caption-19 MuiTypography-colorTextSecondary-42 MuiTypography-alignCenter-32"
        >
          TERMS
        </span>
      </a>
      <a
        className="Component-footerLink-5"
        href="https://termly.io/en/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <span
          className="MuiTypography-root-9 MuiTypography-caption-19 MuiTypography-colorTextSecondary-42 MuiTypography-alignCenter-32"
        >
          PRIVACY
        </span>
      </a>
      <a
        className="Component-footerLink-5"
        href="https://termly.io/en/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <span
          className="MuiTypography-root-9 MuiTypography-caption-19 MuiTypography-colorTextSecondary-42 MuiTypography-alignCenter-32"
        >
          FAIRNESS
        </span>
      </a>
      <a
        className="Component-footerLink-5"
        href="https://termly.io/en/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <span
          className="MuiTypography-root-9 MuiTypography-caption-19 MuiTypography-colorTextSecondary-42 MuiTypography-alignCenter-32"
        >
          ALL PRIZES
        </span>
      </a>
    </div>
  </div>
</div>
`;
